src/Microsoft.Data.Analysis/RowCursor.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.ML;
+using Microsoft.ML.Data;
+
+namespace Microsoft.Data.Analysis
+{
+    internal sealed class RowCursor : DataViewRowCursor
+    {
+        private bool _disposed;
+        private long _position;
+        private readonly DataFrame _dataFrame;
+        internal readonly List<Delegate> _getters;
+        private Dictionary<int, int> _columnIndexToGetterIndex;
+
+        public RowCursor(DataFrame dataFrame, bool[] activeColumns)
+        {
+            Debug.Assert(dataFrame != null);
+            Debug.Assert(activeColumns != null);
+
+            _columnIndexToGetterIndex = new Dictionary<int, int>();
+            _position = -1;
+            _dataFrame = dataFrame;
+            _getters = new List<Delegate>();
+            for (int i = 0; i < Schema.Count; i++)
+            {
+                if (!activeColumns[i])
+                {
+                    continue;
+                }
+
+                Delegate getter = CreateGetterDelegate(i);
+                _getters.Add(getter);
+                Debug.Assert(getter != null);
+                _columnIndexToGetterIndex[i] = _getters.Count - 1;
```

Comment:
I'm not sure I understand why we need `_columnIndexToGetterIndex`.

Reply:
`GetGetter<T>(DataViewSchema.Column column)`  was using `column.Index` to index into `_getters` and return a delegate. However, `_getters` is only populated with "active columns", so `column.Index` doesn't map to `_getter` index if there are hidden columns (or the user has selected columns in `ToDataFrame(SelectColumn: ...)`

Reply:
Do you have a test illustrating the problem?

> However, _getters is only populated with "active columns", so column.Index doesn't map to _getter index if there are hidden columns (or the user has selected columns in ToDataFrame(SelectColumn: ...)

`_getters` is an array with the Length of _all_ the columns. If a column is hidden or not selected, its index into `_getters` will be `null`.

Reply:
Actually, `_getters` is a `List` that has getters only for the active columns. I moved `RowCursor` back to `DataFrame.IDataView.cs`, so this may be easier to spot now. This was an attempt to reduce the size of the `_getters` object when only a few columns are selected. When all the columns are selected, we will end up using more space though. I think most folks will be interested in just a subset of the columns in an IDataView, so I figured the trade-off of using a dictionary made sense.

Reply:
I don't believe we should be making this change with this PR. This change is unrelated to the current work

I reverted the changes to the `DataFrame.IDataView.cs` file and ran all the tests and they still pass. Your reasoning makes it sound like this is intended to be a perf optimization. If so, it should be made separately with perf measurements illustrating the impact the change is making.